src/MongoDB.Driver.Core/Core/Operations/AsyncCursor.cs

```diff
@@ -133,6 +135,27 @@ private int CalculateGetMoreProtocolNumberToReturn()
             return numberToReturn;
         }
 
+        /// <summary>
+        /// Closes the cursor.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        public void Close(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            CloseCursor(cancellationToken);
+            Dispose();
```

Comment:
After discovering the GridFS implementation of the relationship between Close and Dispose I think we need a try/finally here to ensure Dispose gets called even if CloseIfNotAlreadyClosed throws an exception.

Do you agree?

Reply:
yep, I will add this.